src/App.jsx

```diff
@@ -95,17 +97,56 @@ function App() {
     // network. To actually test rinkeby, change setChainID equal to 4 before testing.
     let loadProvider = provider;
 
-    await dispatch(loadAppDetails({ networkID: chainID, provider: loadProvider }));
-    if (address) await dispatch(loadAccountDetails({ networkID: chainID, address, provider: loadProvider }));
-
-    Object.values(BONDS).map(async bond => {
-      await dispatch(calcBondDetails({ bond, value: null, provider: loadProvider, networkID: chainID }));
-    });
+    // NOTE (appleseed): loadDetails() runs three times on every app refresh (every time `connected` is set)...
+    // ... once with address === "" && loadProvider === StaticJsonRpcProvider (set inside of Web3ContextProvider)
+    // ... once with address === "[wallet address]" && loadProvider === StaticJsonRpcProvider (set inside of Web3Context.connect())
+    // ... once with address === "[wallet address]" && loadProvider === Web3Provider (set inside of Web3Context.connect() right after the above line)
+    // So we need to make sure we don't run `loadAccountDetails`, `loadAppDetails` & `calcBondDetails`...
+    // ... below each of the three times state is changed
+    // The below if statements are one way to prevent the 3x runs
+    //
+    // console.log(whichDetails, connected, address, provider);
+
+    // run with loadProvider (backend provider) so that user doesn't need a connected wallet to see app details.
+    if (whichDetails === "app") {
+      loadApp(loadProvider);
+    }
+
+    // don't run unless provider is a Wallet...
+    if (whichDetails === "account" && address && connected) {
+      loadAccount(loadProvider);
+      if (isAppLoaded) return; // Don't need to do anything else if the app is already loaded.
```

Comment:
@lolchocotaco what was the thinking behind this `if (isAppLoaded)` check? I don't see a need for it & on initial paint `isAppLoaded` is always false in every case I have seen, so as a result `loadApp` runs twice (once on line 112 & once on line 120).

Reply:
nevermind. looks like it doesn't matter. dispatch is smart enough not to duplicate the call.